### DISCUSSION TITLE
Accept header not working

### DISCUSSION THREAD

**Discussion by feniljariwala82** [adonisjs/core#3483] [Help] [answered]
According to the documentation: [show](https://docs.adonisjs.com/guides/request#accepts), I tried to add a header in Ajax request. The thing is I want a JSON response from the same controller which renders the index page. Here you can see the header code provided below.

![snip](https://user-images.githubusercontent.com/42708686/149069930-4e9576f6-fbcb-4818-ad71-2724a08ea413.PNG)

After providing **Accept** header in the **Ajax** request. The controller is returning the default view. Can you please guide me on what I am doing wrong here?

**Comment by RomainLanz** [comment, marked as answer]
Hey @feniljariwala82! 👋🏻 

First of all, the correct type should be `application/json`. 
Secondly, you must handle that yourself using [Content Negotiation](https://docs.adonisjs.com/guides/request#content-negotiation).

```ts
Route.get('posts', async ({ request, view }) => {
  const posts = [
    {
      title: 'Adonis 101',
    },
  ]

  switch (request.accepts(['html', 'json'])) {
    case 'html':
      return view.render('posts/index', { posts })
    case 'json':
      return posts
    default:
      return view.render('posts/index', { posts })
  }
})
```

**Reply by feniljariwala82** [reply]
I wrote backend code according to the documentation. I changed accept header, thank you. It's working.
